templates/shift-scheduling/main.go

```diff
@@ -1,370 +1,268 @@
+	m, x := newMIPModel(input, potentialAssignments, potentialAssignmentsPerWorker, demands, opts)
 
-func solver(input schedulingProblem, opts store.Options) (store.Solver, error) {
-	// We start with an empty schedule.
-	schedule := store.New()
-
-	// We define some helper variables that are useful later.
-	nShifts := input.Days * 3
-	workers := model.NewDomain(model.NewRange(1, input.Workers))
-	typeMap := [3]shiftType{"morning", "day", "night"}
-	prefShiftType := map[int]shiftType{}
-	for _, v := range input.Preferences.ShiftType {
-		prefShiftType[int(v.Worker)] = v.Type
+	solver, err := mip.NewSolver("highs", m)
+	if err != nil {
+		return schema.Output{}, err
 	}
-
-	// We track the following variables:
-	// shifts holds one domain for each shift (3 per day in order morning, day,
-	// night) initially it holds all workers available.
-	shifts := store.Repeat(schedule, nShifts, workers)
-
-	// The number of workers active in a schedule.
-	workerCount := store.NewVar(schedule, 0)
-
-	// We also want to maximize worker happiness by fulfilling as many
-	// preferences as possible.
-	happiness := store.NewVar(schedule, 0)
-
-	// As a first step, we can now remove the workers that are unavailable for
-	// certain full days. This results as a new (partial) schedule.
-	changes := []store.Change{}
-
-	for _, v := range input.Unavailable {
-		for _, d := range v.Days {
-			morning := (d - 1) * 3
-			day := morning + 1
-			night := day + 1
-			changes = append(changes,
-				shifts.Remove(morning, []int{int(v.Worker)}),
-				shifts.Remove(day, []int{int(v.Worker)}),
-				shifts.Remove(night, []int{int(v.Worker)}),
-			)
-		}
+	options := mip.NewSolveOptions()
+	options.SetVerbosity(mip.Off)
+	err = options.SetMIPGapRelative(gap)
+	if err != nil {
+		return schema.Output{}, err
+	}
+	err = options.SetMaximumDuration(opts.SolverDuration)
+	if err != nil {
+		return schema.Output{}, err
+	}
+	solution, err := solver.Solve(options)
+	if err != nil {
+		return schema.Output{}, err
 	}
-	schedule = schedule.Apply(changes...)
-
-	// Now we define our objective value. This is the quantity we try to
-	// minimize. It is the weighted sum of workers and happiness. Happiness is
-	// negative as we want to maximize it while minimizing the worker count.
-	workerCountImportance := 10
-	schedule = schedule.Value(func(s store.Store) int {
-		return workerCountImportance*workerCount.Get(s) - happiness.Get(s)
-	})
 
-	// Now we define our constraint containing a propagator. This will remove
-	// workers from domains based upon the workers in other domains.
-	constraint := newBreakConstraint(shifts, 2)
+	output, err := format(solution, input, x, potentialAssignments)
+	if err != nil {
+		return schema.Output{}, err
+	}
 
-	// We invoke the propagator to propagate the initial values.
-	schedule.Propagate(constraint.propagate)
+	return output, nil
+}
 
-	if constraint.isProvenInfeasible(schedule) {
-		return nil, fmt.Errorf("input can not be solved")
+func format(
```

Comment:
I now understand why the output looks different. Can you please look at the other MIP templates and copy what is being used as the `format` function? There is some functionality in `sdk` that already takes care of setting the value, optimality, etc... It would be good to be consistent.

Take this example from the `mip-knapsack` template:

```go
	// Format the solution into the desired output format and add custom
	// statistics.
	output := mip.Format(options, format(input, solution, itemVariables), solution)
	output.Statistics.Result.Custom = mip.DefaultCustomResultStatistics(m, solution)

	return output, nil
}

// format the solution from the solver into the desired output format.
func format(input input, solverSolution mip.Solution, itemVariables map[string]mip.Bool) solution {
	if !solverSolution.IsOptimal() && !solverSolution.IsSubOptimal() {
		return solution{}
	}

	items := make([]item, 0)
	for _, item := range input.Items {
		if solverSolution.Value(itemVariables[item.ItemID]) < 1 {
			continue
		}
		items = append(items, item)
	}

	return solution{
		Items: items,
	}
}
```

You can see that `format` _just_ takes care of translating the `mip.Solution` into the specific `solution` of the template. The rest of the logic is taken care of by these 2 lines:

```go
	output := mip.Format(options, format(input, solution, itemVariables), solution)
	output.Statistics.Result.Custom = mip.DefaultCustomResultStatistics(m, solution)
```

The `mip.Format` function is used...